packages/api/endpoints/granules.js

```diff
@@ -487,15 +488,32 @@ async function del(req, res) {
 async function get(req, res) {
   const {
     knex = await getKnexClient(),
+    collectionPgModel = new CollectionPgModel(),
   } = req.testContext || {};
   const { getRecoveryStatus } = req.query;
   const granuleId = req.params.granuleName;
+  const collectionId = req.params.collectionId;
+
   let granule;
+  let pgCollection;
   try {
-    granule = await getUniqueGranuleByGranuleId(knex, granuleId);
+    if (collectionId) {
+      pgCollection = await collectionPgModel.get(
+        knex, deconstructCollectionId(collectionId)
+      );
+
+      granule = await getGranuleByUniqueColumns(knex, granuleId, pgCollection.cumulus_id);
+    } else {
+      granule = await getUniqueGranuleByGranuleId(knex, granuleId);
+    }
   } catch (error) {
     if (error instanceof RecordDoesNotExist) {
-      return res.boom.notFound('Granule not found');
+      if (collectionId && pgCollection === undefined) {
+        return res.boom.notFound(`No collection found for granuleId ${granuleId} with collectionId ${collectionId}`);
+      }
```

Comment:
Open to discussion here. This throws an error if we can't find the collection referenced by a granule. We could fall back to the old logic and look the granule up by `granuleId` only but that doesn't feel quite right.

Reply:
No, I agree with your inclination here.   

This looks like it should be refactored  to two seperate methods, one that is `getByGranuleId` and one that is `get`, as most of the DRY-ing up achieved by doing this is try/catch scaffolding, and it makes it *much* easier to read/maintain if the function does one thing only instead of option switching.   Particularly if we're going to tear this out soon-ish.

Reply:
Yup, that totally makes sense: https://github.com/nasa/cumulus/pull/2978/commits/7521cdda727d128817985107bf94c157863c52b8

Note that I'm just commenting that `getByGranuleId` is deprecated. The `deprecate` method seems to rely on the functions in question being exported and it doesn't seem worth it to do that here.

Reply:
Yeah, that's fine, this isn't really user-exported anyway.   👍🏻